### ISSUE TITLE
Suggest nnoremap instead of map for lightspeed_last_motion mappings

### ISSUE THREAD

**Issue by gibfahn** [ggandor/lightspeed.nvim#101]
I added the mapping lines to my vimrc as suggested in the doc: 

https://github.com/ggandor/lightspeed.nvim/blob/02c3138195c1608fa07b5fec73c3a0d3c0f9bc3e/doc/lightspeed.txt#L383-L384

and they overrode the default stage mappings in the vim-fugitive bindings:

https://github.com/tpope/vim-fugitive/blob/2064312ad7bb80050baf9acbdfb7641162919e53/doc/fugitive.txt#L269

I switched to this, and it seemed to still work as well:

```vim
nnoremap <expr> ; g:lightspeed_last_motion == 'sx' ? "<Plug>Lightspeed_;_sx" : "<Plug>Lightspeed_;_ft"
nnoremap <expr> , g:lightspeed_last_motion == 'sx' ? "<Plug>Lightspeed_,_sx" : "<Plug>Lightspeed_,_ft"
```

**Comment by ggandor** [comment]
> they overrode the default stage mappings in the vim-fugitive bindings

I don't understand, what do `;`/`,` have to do with `s`/`U`/etc.? I think it's rather the default `s` key of Lightspeed. It seems Fugitive does the same thing as Lightspeed, and maps its default keys only if they're not already remapped - I guess Lightspeed comes before Fugitive in your config, so it prevents the latter to map `s`. Try to switch the order, and map `Lightspeed_s/S` to something else then.

> it seemed to still work as well

Yep, but it will only work for instant-repeat then (repeating a fresh search, while we're still executing), and only because of a trick we're doing in the script (tl;dr, we actually evaluate the right hand side of the mapping in which the input key is used, and then simply check whether it's == the string `<Plug>Lightspeed...`). However, if you do this change, you cannot start a repeated search (sometime later) with `;`/`,`.

**Comment by ggandor** [comment]
Any remarks / persisting problems? Shall I close this?

**Comment by gibfahn** [comment]
Hi, sorry for the delay.

>I don't understand, what do ;/, have to do with s/U/etc.? I think it's rather the default s key of Lightspeed that is in conflict, isn't it?

I was pretty sure I made this change and nothing else, and fugitive started working again, but now I try it (having rewritten my config in lua in the meantime) this doesn't seem to make a difference, so happy to close.

> If you do this change, you cannot start a repeated search (sometime later) with ;/, anymore.

You're totally right, thanks for explaining, I've reverted that change.

>It seems Fugitive does the same thing as Lightspeed, and maps its default keys only if they're not already remapped - I guess Lightspeed comes before Fugitive in your config, so it prevents the latter to map s. 

Interestingly in my current config, light speed (https://github.com/gibfahn/dot/blob/1973f6ca1ef609b9e641c17b1ca6f7cccf93a1fc/dotfiles/.config/nvim/init.lua#L363) comes before fugitive (https://github.com/gibfahn/dot/blob/1973f6ca1ef609b9e641c17b1ca6f7cccf93a1fc/dotfiles/.config/nvim/init.lua#L381) due to alphabetical sorting, and things are still working, even with this change reverted.

If I run `nmap s` in a fugitive buffer (`filetype=fugitive`) then I see both mappings (and the fugitive one takes precedence):

```vim script
:nmap s
n  s           &@:<C-U>execute <SNR>137_Do('Stage',0)<CR>
n  s             <Plug>Lightspeed_s
```

But outside I only have the Lightspeed one, and everything works as expected.

>You can switch the order, and map Lightspeed_s/S to something else, but of course it would be much saner to map Fugitive s to something else instead, as that is a much less frequently used command.

I find `s` for `stage` in fugitive to be a really obvious command, I've never wanted to sneak while in a fugitive buffer, and fugitive is one of the most commonly used plugins out there, so I don't think this is an obvious solution 😁 .
